R/makeTreeSummarizedExperimentFromBiom.R

```diff
@@ -224,3 +244,77 @@ makeTreeSummarizedExperimentFromBiom <- function(obj, ...){
     }
     return(colname)    
 }
+
+# Detect and clean non wanted characters from Taxonomy data if needed.
+.detect_taxa_artifacts_and_clean <- function(x, patterns) {
+    
+    # No cleaning
+    if (is.null(patterns)) {
+        warning("No Taxonomy cleaning was done to the data.",
+                call. = FALSE)
+    }
+    # Automatic cleaning
+    else if (patterns=="auto") {
+        # General regex pattern
+        PATTERN <- "[[:alnum:]]|-|_|\\[|\\]|,|;\\||[[:space:]]"
+        patterns <- .detect_taxa_artifacts(x, PATTERN, invert=TRUE)
+        # No artifacts found
+        if (patterns=="") {
+            warning("No artifacts found in the Taxonomy data.",
+                    call. = FALSE)
+        # Clean from artifacts
+        } else {
+            x <- .clean_from_artifacts(x, patterns)
+            warning("The following artifacts: '", patterns, "' were cleaned from 
+                    the Taxonomy data.",
+                    call. = FALSE)
+        }
+    # Clean with the character or regex provided
+    } else {
+        pattern <- .detect_taxa_artifacts(x, patterns=patterns)
+        # patterns provided not found
+        if (pattern=="") {
+            warning("The '", patterns, "' provided at 'cleanTaxaPattern' were
+                    not found in the Taxonomy data.",
+                    call. = FALSE)
+        # patterns found and cleaned
+        } else {
+            x <- .clean_from_artifacts(x, pattern)
+            # warn what was cleaned
+            warning("The following artifacts: '", pattern, "' were cleaned from 
+                    the Taxonomy data.",
+                    call. = FALSE)
```

Comment:
see above - this seems repetitive compared to the above, is there a way to call the same detect & clean code just once?

Reply:
Yes, I will update.